src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -502,7 +502,7 @@ private void CreateRequestBuilderClass(CodeNamespace currentNamespace, OpenApiUr
         foreach (var child in currentNode.Children)
         {
             var propIdentifier = child.Value.GetNavigationPropertyName(config.StructuredMimeTypes);
-            var propType = child.Value.DoesNodeBelongToItemSubnamespace() ? propIdentifier + itemRequestBuilderSuffix : propIdentifier + requestBuilderSuffix;
+            var propType = child.Value.DoesNodeBelongToItemSubnamespace() ? child.Value.GetNavigationPropertyName(config.StructuredMimeTypes, itemRequestBuilderSuffix) : child.Value.GetNavigationPropertyName(config.StructuredMimeTypes, requestBuilderSuffix);
```

Comment:
What do you think of this minor refactor? 🤔
```csharp
var propType = child.Value.GetNavigationPropertyName(config.StructuredMimeTypes, child.Value.DoesNodeBelongToItemSubnamespace() ? itemRequestBuilderSuffix : requestBuilderSuffix)
```

Reply:
Resolved in latest update.